subsys/net/l2/ieee802154/ieee802154.c

```diff
@@ -387,8 +387,9 @@ static enum net_verdict ieee802154_recv(struct net_if *iface, struct net_pkt *pk
 
 	if (fs->fc.frame_type == IEEE802154_FRAME_TYPE_BEACON) {
 		verdict = ieee802154_handle_beacon(iface, &mpdu, net_pkt_ieee802154_lqi(pkt));
-		if (verdict == NET_OK) {
+		if (verdict == NET_CONTINUE) {
 			net_pkt_unref(pkt);
+			return NET_OK;
```

Comment:
This changes (look only in the PR) makes me think that there is no central point to `unref` the frames.
Is there a clear way to `unref` a frame that can be used in general? I mean, each time a NET_OK will be returned (frame processed) the package should be `unref`. Is my assumption that correct?

Reply:
Yes, exactly. That is a assumption valid throughout the net stack and that's also the very reason why I changed the semantics here - previously the internal methods would return NET_OK, although they did NOT unref the packages.

The correct way throughout the stack is:
* return NET_CONTINUE/NET_DROP: the upper layers are responsible for unref'ing the packet
* return NET_OK: upper layers will assume that the packet has been consumed and unref'ed

We still have a (well documented) exception in the driver API (ieee802154_handle_ack) - it is assumed that drivers will always unref the packet, even if this method returns NET_OK, that's for backwards compatibility and has a TODO attached to it.